### PR TITLE
ADX-283 Remove unnecessary command creating validation badges too early.

### PR DESCRIPTION
**Pull request by jonathansberry**
I havn't been able to confirm this yet. Everytime I come to try and test this something else strange that I don't understand is thrown up.  I've spent today trying to get some tests going for this because it's completely untenable. 

Tests are going to take quite a while to develop though. 

Not sure how to proceed since I need to spend tomorrow on the UNAIDS presentation.

### CODE REVIEW

**Comment by jonathansberry** [comment]
@tomeksabala  I think after sitting down for 30 mins with a fresh head and a fresh dev env, I have actually fixed the issue with the second commit.  Testing works well locally now. 

I have just force pushed to clean this PR up a bit. You may want to do a hard reset if you have pulled the branch down locally. I have moved my test stubs to a separate branch ADX-283-validation tests, and opened a draft PR.

File: ckanext/validation/plugin.py [hunk 1]
```diff
@@ -188,6 +186,10 @@ def after_create(self, context, data_dict):
             for resource in data_dict.get(u'resources', []):
                 self._handle_validation_for_resource(context, resource)
         else:
+            for plugin in p.PluginImplementations(IDataValidation):
+                if not plugin.can_validate(context, data_dict):
+                    log.debug('Skipping validation for resource {}'.format(data_dict['id']))
+                    return
```
Comment (jonathansberry) [inline ckanext/validation/plugin.py:192]: This code checks for plugins implementing IDataValidation.  ckanext-unaids does this, and simply checks that the resourrce has the field "schema" in it, before continuing with validation. This was missing previously, meaning that whenever the logic landed here, validation was being triggered irrespective of whether the resource has a schema defined or not.

File: ckanext/validation/plugin.py [hunk 2]
```diff
@@ -252,7 +254,6 @@ def before_update(self, context, current_resource, updated_resource):
             needs_validation = True
 
         if needs_validation:
-            updated_resource['validation_status'] = 'unkown'
```
Comment (jonathansberry) [inline ckanext/validation/plugin.py:255]: Setting validation_status now happens using resource_patch from the resource_validation_run action.  It no longer needs to happen here.

File: ckanext/validation/plugin.py [hunk 3]
```diff
@@ -171,8 +171,6 @@ def _process_schema_fields(self, data_dict):
 
     def before_create(self, context, data_dict):
         updated_data_dict = self._process_schema_fields(data_dict)
-        if updated_data_dict.get('schema'):
-            updated_data_dict['validation_status'] = 'unknown'
```
Comment (jonathansberry) [inline ckanext/validation/plugin.py:175]: Setting validation_status now happens using resource_patch from the resource_validation_run action.  It no longer needs to happen here.

File: ckanext/validation/plugin.py (hunk 1 above)
Comment (tomeksabala) [inline ckanext/validation/plugin.py:192]: It's a code dup from: `_handle_validation_for_resource`
We should either run here `self._handle_validation_for_resource` and decorate it if needed or create a new `_handle_special_validation_for_resource` which would share some private code with `_handle_validation_for_resource`

This solution has a lot of technical dept. We might be too busy atm however to handle it right now.

**Review by tomeksabala — APPROVED** [review]

**Pull request merged**